filesystem/fat32/fat32.go

```diff
@@ -631,7 +631,7 @@ func (fs *FileSystem) writeDirectoryEntries(dir *Directory) error {
 	}
 	extraClusters := len(b)/(int(fs.bootSector.biosParameterBlock.dos331BPB.dos20BPB.sectorsPerCluster)*fs.bytesPerCluster) - len(clusterList)
 	if extraClusters > 0 {
-		clusters, err := fs.allocateSpace(uint64(extraClusters), clusterList[len(clusterList)-1])
+		clusters, err := fs.allocateSpace(uint64(len(b)), clusterList[0])
```

Comment:
I assume this was what you mean when you wrote:

> function needs to be called with a file size and the beginning of the current chain (otherwise the size delta is not computable)

Can you explain these?

* Clearly, passing the `extraClusters` was wrong, since it should pass the bytes required. But this would pass it the entire `b`, when all we need is the excess. 
* Why do we pass it the previous as the first, i.e. `clusterList[0]`, and not the last?

Reply:
It is possible to call with len and let it load the chain, or call with the delta and the last sector. I'll check if I can simplify this, I think I might have overshoot the goal a bit trying to fix this